yb-voyager/src/migassessment/sizing.go

```diff
@@ -652,6 +795,128 @@ func getSourceMetadata(sourceDB *sql.DB) ([]SourceDBMetadata, []SourceDBMetadata
 	return sourceTableMetadata, sourceIndexMetadata, totalSourceDBSize, nil
 }
 
+/*
+checkAndFetchIndexes checks for indexes associated with a specific database table and fetches their metadata.
+It iterates through a slice of index metadata and selects indexes that belong to the specified table by comparing
+their parent table names. The function returns a slice containing metadata for indexes associated with the table
+and the total size of those indexes.
+Parameters:
+
+	table: Metadata for the database table for which indexes are to be checked.
+	indexes: A slice containing metadata for all indexes in the database.
+
+Returns:
+
+	[]SourceDBMetadata: Metadata for indexes associated with the specified table.
+	float64: The total size of indexes associated with the specified table.
+	int64 : sum of read ops per second for all indexes of the table
+	int64 : sum of write ops per second for all indexes of the table
+*/
+func checkAndFetchIndexes(table SourceDBMetadata, indexes []SourceDBMetadata) ([]SourceDBMetadata, float64, int64, int64) {
+	indexesOfTable := make([]SourceDBMetadata, 0)
+	var indexesSizeSum float64 = 0
+	var cumulativeSelectOpsPerSecIdx int64 = 0
+	var cumulativeInsertOpsPerSecIdx int64 = 0
+	for _, index := range indexes {
+		if index.ParentTableName.Valid && (index.ParentTableName.String == (table.SchemaName + "." + table.ObjectName)) {
+			indexesOfTable = append(indexesOfTable, index)
+			indexesSizeSum += index.Size
+			cumulativeSelectOpsPerSecIdx += index.ReadsPerSec
+			cumulativeInsertOpsPerSecIdx += index.WritesPerSec
+		}
+	}
+
+	return indexesOfTable, indexesSizeSum, cumulativeSelectOpsPerSecIdx, cumulativeInsertOpsPerSecIdx
+}
+
+/*
+getReasoning generates a string describing the reasoning behind a recommendation for an instance type.
+It considers the characteristics of colocated and sharded objects.
+Parameters:
+  - recommendation: An IntermediateRecommendation struct containing information about the recommended instance type.
+  - shardedObjects: A slice of SourceDBMetadata structs representing objects that need to be sharded.
+  - colocatedObjects: A slice of SourceDBMetadata structs representing objects that can be colocated.
+
+Returns:
+  - A string describing the reasoning behind the recommendation.
+*/
+func getReasoning(recommendation IntermediateRecommendation, shardedObjects []SourceDBMetadata, colocatedObjects []SourceDBMetadata) string {
+	// Calculate size and throughput of colocated objects
+	colocatedObjectsSize, colocatedReads, colocatedWrites := getObjectsSize(colocatedObjects)
+	reasoning := fmt.Sprintf("Recommended instance type with %v vCPU and %vGiB memory could fit ",
+		recommendation.VCPUsPerInstance, recommendation.VCPUsPerInstance*recommendation.MemoryPerCore)
+
+	// Add information about colocated objects if they exist
+	if len(colocatedObjects) > 0 {
+		reasoning += fmt.Sprintf("%v object(s) with %0.4fGB size and throughput requirement of %v reads/sec"+
```

Comment:
0.4fGB -> 0.2f GB

Reply:
the current one is correct in my opinion for very small tables. Example: if a table is of size 1MB, it will show as 0.00 GB if we do 0.2f. Added the space though.